### DISCUSSION TITLE
[question] Options在X6的interface中是什么意思呢，有什么功能？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [antvis/X6#4403] [Q&A] [answered]
我检验X6代码的时候，经常有看到：Options相关的interface
https://github.com/antvis/X6/blob/master/packages/x6/src/model/edge.ts#L895
https://github.com/antvis/X6/blob/master/packages/x6/src/view/cell.ts#L755
https://github.com/antvis/X6/blob/master/packages/x6/src/view/edge.ts#L2388
https://github.com/antvis/X6/blob/master/packages/x6/src/view/node.ts#L1135

请问，Options在X6的interface中是什么意思呢，有什么功能？

**Comment by resetsix** [comment]
`interface`不就是[ts 类型声明](https://www.typescriptlang.org/docs/handbook/interfaces.html)吗

类似于 Java 中的「接口」

**Reply by liaodalin19903** [reply]
您好，指的是Options。

**Reply by resetsix** [reply, marked as answer]
> 您好，指的是Options。

我尝试用我自己粗略的理解来解释（也有可能不正确）。

你在很多地方看见的`Options`其实都是各自独立的入参配置，例如 node 有自己的 Options，edge 有自己的 Options，只是统一叫「Options」这个命名而已。

从你给出的链接来举例，NodeView 和 EdgeView 命名空间分别对应 node 和 edge 的类型声明，各自都声明了一个 Options 来继承[CellView.Options](https://github.com/antvis/X6/blob/003491bd17d6fb5677f561035ddfc678bc1b2d73/packages/x6/src/view/cell.ts#L755)内的所有类型。而[Cell](https://x6.antv.antgroup.com/api/model/cell)是什么呢？

> Cell 是 [Node](https://x6.antv.antgroup.com/zh/docs/api/model/node) 和 [Edge](https://x6.antv.antgroup.com/zh/docs/api/model/edge) 的基类，包含节点和边的通用属性和方法定义，如属性样式、可见性、业务数据等，并且在实例化、样式定制、默认选项、自定义选项等方面具有相同的行为。

也就是说Options并没有拓展任何类型，它也可以理解为父类型的别名（使用 interface 的原因是interface 可以继承和拓展，而 type 更适合联合类型不具备继承和拓展的特性。）

<details>
<summary>点击展开/关闭</summary>

<img width="461" alt="图片" src="https://github.com/user-attachments/assets/06dbe5e9-62c3-45ec-bc24-ac14975afdd4">

<img width="453" alt="图片" src="https://github.com/user-attachments/assets/55faa99c-7826-4e70-b609-100ec4d654f3">

</details>

再看下图的 Options，它继承了ToolsView.ToolItem.Options，并且拓展了自己的类型，而它是 Boundary（边界）的类型。

> [boundary](https://x6.antv.antgroup.com/api/registry/edge-tool#boundary)通常指的是一个图形元素（如节点或边）的外部轮廓或边界框。

<details>
<summary>点击展开/关闭</summary>

<img width="518" alt="图片" src="https://github.com/user-attachments/assets/2655b0f3-8ffc-4c6b-ba5e-7a93a307c95c">

<img width="1581" alt="图片" src="https://github.com/user-attachments/assets/c7f5d95a-6232-44b7-8c10-8aa58afa9d4f">

</details>

所以不必要在意 Options 是什么，调用的 node、edge、graph api 就是 Options 的内容（来自于父类继承和自身拓展）。

感觉还是没有讲清楚。
